Makefile

```diff
@@ -121,13 +141,39 @@ $(BUILDDIR)/tctl:
 	GOOS=$(OS) GOARCH=$(ARCH) $(CGOFLAG) go build -tags "$(PAM_TAG) $(FIPS_TAG) $(BPF_TAG)" -o $(BUILDDIR)/tctl $(BUILDFLAGS) ./tool/tctl
 
 .PHONY: $(BUILDDIR)/teleport
-$(BUILDDIR)/teleport: ensure-webassets
+$(BUILDDIR)/teleport: ensure-webassets bpf-bytecode
+	@echo $(CGOFLAG)
 	GOOS=$(OS) GOARCH=$(ARCH) $(CGOFLAG) go build -tags "$(PAM_TAG) $(FIPS_TAG) $(BPF_TAG)" -o $(BUILDDIR)/teleport $(BUILDFLAGS) ./tool/teleport
 
 .PHONY: $(BUILDDIR)/tsh
 $(BUILDDIR)/tsh:
 	GOOS=$(OS) GOARCH=$(ARCH) $(CGOFLAG) go build -tags "$(PAM_TAG) $(FIPS_TAG)" -o $(BUILDDIR)/tsh $(BUILDFLAGS) ./tool/tsh
 
+ifeq ("$(with_bpf)","yes")
+$(BPF_BUILDDIR):
+	mkdir -p $(BPF_BUILDDIR)
+
+# Build BPF code
+$(BPF_BUILDDIR)/%.bpf.o: bpf/%.bpf.c $(wildcard bpf/*.h) | $(BPF_BUILDDIR)
+	$(CLANG) -g -O2 -target bpf -D__TARGET_ARCH_$(KERNEL_ARCH) $(INCLUDES) $(CLANG_BPF_SYS_INCLUDES) -c $(filter %.c,$^) -o $@
+	$(LLVM_STRIP) -g $@ # strip useless DWARF info
```

Comment:
Can you add a target that can be used to generate `vmlinux.h` as well? https://github.com/iovisor/bcc/tree/master/libbpf-tools#vmlinuxh-generation

Reply:
I can add a target but I still think we should manually generate it and commit it. Otherwise they'll a reliance on `bpftool` at compile time (an extra dep).